src/Microsoft.SqlTools.ServiceLayer/SqlAssessment/SqlAssessmentService.cs

```diff
@@ -0,0 +1,449 @@
+                {
+                    await requestContext.SendError(SR.QueryServiceQueryInvalidOwnerUri);
+                    return;
+                }
+
+                ConnectParams connectParams = new ConnectParams
+                                                  {
+                                                      OwnerUri = randomUri,
+                                                      Connection = connInfo.ConnectionDetails,
+                                                      Type = ConnectionType.Default
+                                                  };
+
+                var workTask = CallAssessmentEngine<TResult>(
+                    requestParams,
+                    connectParams,
+                    randomUri,
+                    assessmentFunction)
+                    .ContinueWith(tsk =>
+                        {
+                            var f = tsk.IsCompleted;
+                            requestContext.SendResult(tsk.Result).Wait();
+                        });
+
+                ActiveRequests.TryAdd(randomUri, workTask);
+            }
+            catch (Exception ex)
+            {
+                if (ex is StackOverflowException || ex is OutOfMemoryException)
+                {
+                    throw;
+                }
+
+                await requestContext.SendError(ex.ToString());
+            }
+        }
+
+        /// <summary>
+        /// Constructs a <see cref="SqlObjectLocator"/> for specified database.
+        /// </summary>
+        /// <param name="server">Target server locator.</param>
+        /// <param name="databaseName">Target database name.</param>
+        /// <returns>Returns a locator for target database.</returns>
+        private static SqlObjectLocator GetDatabaseLocator(SqlObjectLocator server, string databaseName)
+        {
+            var db = new SqlObjectLocator();
+
+            db.Connection = server.Connection;
+            db.EngineEdition = server.EngineEdition;
+            db.Name = databaseName;
+            db.Platform = server.Platform;
+            db.ServerName = server.Name;
+            db.Type = SqlObjectType.Database;
+            db.Urn = $"{server.Name}/{databaseName}";
+            db.Version = server.Version;
+
+            return db;
+        }
+
+        #endregion
+
+        #region Private Helpers
+
+        /// <summary>
+        /// Converts numeric <paramref name="representation"/> of engine edition
+        /// returned by SERVERPROPERTY('EngineEdition').
+        /// </summary>
+        /// <param name="representation">
+        /// A number returned by SERVERPROPERTY('EngineEdition').
+        /// </param>
+        /// <returns>
+        /// Returns a <see cref="SqlEngineEdition"/>
+        /// corresponding to the <paramref name="representation"/>.
+        /// </returns>
+        private static SqlEngineEdition GetEngineEdition(int representation)

```

Comment:
What is this SqlEngineEdition and why is this cast necessary?

Reply:
This is a value used by the assessment engine. I'm replacing this conversion function with a less cryptic one.